shopfloor_reception_packaging_dimension/services/reception.py

```diff
@@ -0,0 +1,233 @@
+
+        It helps for extending the module to add more dimension fields to be updated.
+
+        """
+        for dimension in dimensions:
+            if local_table[dimension] is None:
+                if dimension in self.work.dimension_to_update.keys():
+                    self.work.dimension_to_update.pop(dimension)
+            else:
+                self.work.dimension_to_update[dimension] = local_table[dimension]
+
+    def set_packaging_dimension(
+        self,
+        picking_id,
+        selected_line_id,
+        packaging_id,
+        height=None,
+        length=None,
+        width=None,
+        weight=None,
+        qty=None,
+        barcode=None,
+        cancel=False,
+    ):
+        """Set the dimension on a product packaging.
+
+        If the user cancel the dimension update we still propose the next
+        possible packgaging.
+
+        Transitions:
+            - set_packaging_dimension: if more packaging needs dimension
+            - set_quantity: otherwise
+        """
+        self._store_dimension_to_update(
+            ["height", "length", "width", "weight", "barcode", "qty"], locals()
+        )
+        picking = self.env["stock.picking"].browse(picking_id)
+        selected_line = self.env["stock.move.line"].browse(selected_line_id)
+        packaging = self.env["product.packaging"].sudo().browse(packaging_id)
+        message = None
+        next_packaging = None
+        if not packaging:
+            message = self.msg_store.record_not_found()
+        elif not cancel and self._check_dimension_to_update():
+            self._update_packaging_dimension(packaging)
+            message = self.msg_store.packaging_dimension_updated(packaging)
+        if packaging:
+            next_packaging = self._get_next_packaging_to_set_dimension(
+                selected_line.product_id, packaging
+            )
+        if not next_packaging:
+            return self._response_for_set_quantity(
+                picking, selected_line, message=message
+            )
+        return self._response_for_set_packaging_dimension(
+            picking, selected_line, next_packaging, message=message
+        )
+
+    def _get_packaging_fields_to_update(self):
+        return ["height", "length", "width", "weight", "barcode", "qty"]
+
+    def _check_dimension_to_update(self):
+        """Return True if there is any dimension that needs to be updated on the packaging."""
+        dimension_to_check = self._get_packaging_fields_to_update()
+        return any(
+            [
+                self.work.dimension_to_update.get(dim, False)
+                for dim in dimension_to_check
+            ]
+        )
+
+    def _update_packaging_dimension(self, packaging):
+        """Update dimension on the packaging."""
+        dimension_to_update = self._get_packaging_fields_to_update()
+        fields_conv_map = {"weight": "max_weight", "length": "packaging_length"}
```

Comment:
Could you maybe move this also to an own method, this would make it easier to hook.

Reply:
Done